code/go/0chain.net/smartcontract/partitions/random.go

```diff
@@ -205,6 +207,14 @@ func (rs *randomSelector) addPartition() PartitionItemList {
 		newPartition = &itemList{
 			Key: rs.partitionKey(rs.NumPartitions),
 		}
+	} else if rs.ItemType == ItemBlobber {
+		newPartition = &blobberItemList{
+			Key: rs.partitionKey(rs.NumPartitions),
+		}
+	} else if rs.ItemType == ItemBlobberReward {
```

Comment:
but @peterlimg partitions are of different for different itemtype
hence using it

Reply:
NVM, we can refactor this later. Anyway, I think each type of partition should has it's own RandomSelector instance, therefore, each RandomSelector can run related interface to generate its own type of partition item list instead of using the `if else..` here.

Reply:
got it.
@peterlimg can you create an issue on the exact thought you have in mind. We can pick it up later